variables.tf

```diff
@@ -0,0 +1,238 @@
+    source                      = optional(string, "")
+    substitution                = optional(string, "")
+  }))
+  default = []
+}
+
+# Datadog logging
+variable "logging_datadog" {
+  description = "Datalog connection and logging information."
+  type = object({
+    name = string
+
+    token  = optional(string, "")
+    format = optional(string, "")
+    region = optional(string, "EU")
+  })
+  sensitive = true
+  default   = null
+}
+
+variable "logging_datadog_token" {
+  description = "API key from Datadog. Used to pull value from Terraform Cloud."
+  type        = string
+  sensitive   = true
+  default     = ""
+}
+
+# Custom VCL snippets
+variable "snippets" {
+  description = "Custom VCL snippets to insert."
+  type = list(object({
+    name     = string
+    content  = string
+    type     = string
+    priority = optional(number, 100)
+  }))
+  default = []
+}
+
+# Custom response objects
+variable "responses" {
+  description = "Custom response objects. Automatically generates condition objects."
+  type = list(object({
+    name               = string
+    content            = string
+    content_type       = string
+    condition          = string
+    condition_type     = string
+    condition_priority = optional(number, 10)
+    response           = string
+    status             = string
+  }))
+  default = []
+}
+
+# IP block lists
+
+variable "ip_blocklist" {
+  description = "List of IP CIDR blocks to deny access."
+  type        = list(string)
+  default     = []
+
+  validation {
+    condition     = can([for s in var.ip_blocklist : regex("^([0-9]{1,3}\\.){3}[0-9]{1,3}(\\/([0-9]|[1-2][0-9]|3[0-2]))?$", s)])
+    error_message = "Each list item must be in a CIDR block format. Example: [\"10.106.108.0/25\"]."
+  }
+}
+
+variable "ip_blocklist_acl_name" {
+  description = "Name for the ACL responsible for holding all the blocked IP ranges."
+  type        = string
+  default     = "IP Block list"
+}
+
+variable "products" {
+  description = "List of additional Fastly products to enable on the service."
+  type = object({
+    brotli_compression = optional(bool, false)
+    domain_inspector   = optional(bool, false)
+    image_optimizer    = optional(bool, false)
+    origin_inspector   = optional(bool, false)
+    websockets         = optional(bool, false)
+  })
+  default = {
+    brotli_compression = false
+    domain_inspector   = false
+    image_optimizer    = false
+    origin_inspector   = false
+    websockets         = false
+  }
+}
+
+# AS block lists
+
+variable "as_blocklist" {
+  description = "List of Autonomous Systems (AS) to block."
+  type        = list(number)
+  default     = []
+}
+
+variable "as_blocklist_name" {
+  description = "Name of the AS blocklist"
+  type        = string
+  default     = "Blocked AS Numbers"
+}
+
+variable "as_request_blocklist" {
+  description = "List of Autonomous Systems (AS) to block from making explicit requests."
```

Comment:
What is an explicit request here?
We may want to define it, or make the name cleared

Reply:
Good point. Let me tweak it a bit.